Gruntfile.js

```diff
@@ -16,13 +16,25 @@ var _ = require("lodash");
 var path = require("path");
 var execSync = require("child_process").execSync;
 
+var getFromExec = function (command) {
+    var result;
+    try {
+        result = execSync(command);
+    } catch (e) {
+        console.log("Error executing command: " + command);
+        console.log(e.stack);
+        result = "unknown\n";
+    }
+    return result;
+};
+
 module.exports = function (grunt) {
 
     // Project configuration.
     grunt.initConfig({
         pkg: grunt.file.readJSON("package.json"),
-        revision: execSync("git rev-parse --verify --short HEAD"),
-        branch: execSync("git rev-parse --abbrev-ref HEAD"),
+        revision: getFromExec("git rev-parse --verify --short HEAD"),
```

Comment:
With the above fixes, this could become getFromExec( ...., {silent: true, defaultValue: "Unknown revision, not within a git repository"})